ugc/types.hpp

```diff
@@ -30,14 +31,14 @@ struct TranslationKey
   bool operator==(TranslationKey const & rhs) const { return m_key == rhs.m_key; }
   bool operator<(TranslationKey const & rhs) const { return m_key < rhs.m_key; }
 
-  friend std::string DebugPrint(TranslationKey const & key)
-  {
-    return "TranslationKey [ " + key.m_key + " ]";
-  }
-
   std::string m_key;
 };
 
+std::string DebugPrint(TranslationKey const & key)
```

Comment:
This method should be marked as inline, otherwise linker may complain about multiple definition. But I suggest you to get rid of DebugPrint and use DECLARE_VISITOR_AND_DEBUG_PRINT instead.

Reply:
заменил